esmf_regrid/schemes.py

```diff
@@ -28,7 +28,89 @@ def _get_coord(cube, axis):
     return coord
 
 
-def _cube_to_GridInfo(cube, center=False, resolution=None):
+def _get_mask(cube):
+    src_x, src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
+
+    if cube.coord_dims(src_x) == cube.coord_dims(src_y):
+        slices = cube.slices([src_x])
+    else:
+        slices = cube.slices([src_x, src_y])
+    data = next(slices).data
```

Comment:
To be honest, I kept the original code because otherwise I would have to figure out how to handle the dimension ordering again. I think there is some code going on behind the scenes with cube dims for 2D coords that I want to use here. I suppose I could write this myself to be more explicit, for the time being though, I wanted to be sure the code was consistent with what I had written.